services/hh/processor.go

```diff
@@ -284,3 +306,41 @@ func (p *Processor) PurgeOlderThan(when time.Duration) error {
 	}
 	return nil
 }
+
+func (p *Processor) DeleteOlderThan(when time.Duration) error {
+	p.mu.Lock()
+	defer p.mu.Unlock()
+
+	deletedQueues := make([]uint64, 0)
+	for nodeID, queue := range p.queues {
+		// Only delete queues for inactive nodes.
+		ni, err := p.metastore.Node(nodeID)
+		if err != nil {
+			return err
+		}
+		if ni != nil {
+			continue
+		}
+
+		last, err := queue.LastModified()
+		if err != nil {
+			return err
+		}
+		if last.Before(time.Now().Add(-when)) {
+			// Close and remove the queue.
+			if err := queue.Close(); err != nil {
+				return err
+			}
+			if err := os.RemoveAll(queue.dir); err != nil {
```

Comment:
Yeah, I thought about that, but wasn't sure. Should `Remove()` work on an open-queue? Should the queue be closed first? What are the semantic of this command?

All these reasons made me fall back on having this operation occur outside of the queue. (I started with having it as a method on the queue, as you suggest). I don't feel super strongly about this, and wasn't too happy with the code acting on `path`, technically an unexported variable (but fine obviously within the package).

I'm happy with whatever is clearest to the next guy.